src/MobileRedirection.php

```diff
@@ -165,21 +177,59 @@ public function redirect() {
 			}
 
 			// Add a link to the footer to allow for navigation to the AMP version.
-			add_action( 'wp_footer', [ $this, 'add_mobile_version_switcher_link' ] );
+			$this->add_mobile_switcher_footer_hooks( false );
 		} else {
 			if ( ! $js && $this->is_redirection_disabled_via_cookie() ) {
 				$this->set_mobile_redirection_disabled_cookie( false );
 			}
 
-			add_filter( 'amp_to_amp_linking_element_excluded', [ $this, 'filter_amp_to_amp_linking_element_excluded' ], 100, 2 );
-			add_filter( 'amp_to_amp_linking_element_query_vars', [ $this, 'filter_amp_to_amp_linking_element_query_vars' ], 10, 2 );
+			$this->add_a2a_linking_hooks();
 
 			// Add a link to the footer to allow for navigation to the non-AMP version.
-			add_action( 'wp_footer', [ $this, 'add_mobile_version_switcher_link' ] );
+			$this->add_mobile_switcher_footer_hooks();
+		}
+	}
+
+	/**
+	 * Add mobile version switcher link/styles when mobile redirection is not enabled.
+	 */
+	public function maybe_add_mobile_switcher_hooks() {
+		if ( $this->is_mobile_request() ) {
+			$this->add_mobile_switcher_head_hooks();
+			$this->add_a2a_linking_hooks();
+			$this->add_mobile_switcher_footer_hooks();
+		}
+	}
+
+	/**
+	 * Add mobile version switcher head hooks.
+	 */
+	private function add_mobile_switcher_head_hooks() {
+		add_action( 'wp_head', [ $this, 'add_mobile_version_switcher_styles' ] );
+		add_action( 'amp_post_template_head', [ $this, 'add_mobile_version_switcher_styles' ] ); // For legacy Reader mode theme.
+	}
+
+	/**
+	 * Add mobile version switcher footer hooks.
+	 *
+	 * @param bool $legacy Whether to add mobile version switcher for legacy Reader mode theme.
+	 */
+	private function add_mobile_switcher_footer_hooks( $legacy = true ) {
+		add_action( 'wp_footer', [ $this, 'add_mobile_version_switcher_link' ] );
+
+		if ( $legacy ) {
 			add_action( 'amp_post_template_footer', [ $this, 'add_mobile_version_switcher_link' ] ); // For legacy Reader mode theme.
 		}
```

Comment:
I don't think the `$legacy` parameter is needed. It's not present in the `head` version. The parameter just adds complexity.

```suggestion
	 */
	private function add_mobile_switcher_footer_hooks() {
		add_action( 'wp_footer', [ $this, 'add_mobile_version_switcher_link' ] );
		add_action( 'amp_post_template_footer', [ $this, 'add_mobile_version_switcher_link' ] ); // For legacy Reader mode theme.
```